nautobot/ipam/factory.py

```diff
@@ -111,3 +107,199 @@ def prefix(self, n):
             return f"2003:0{n:X}::/32"
 
         raise RuntimeError(f"Don't know how to pick an address for RIR {self.rir.name}")
+
+
+def random_route_distinguisher():
+    """
+    The RD is an arbitrary unique string up to VRF_RD_MAX_LENGTH (21) characters with no explicit validation.
+
+    However by convention it will be one of:
+    - "<2-byte ASN>:<4-byte integer>"
+    - "<IPv4 address>:<2-byte integer>"
+    - "<4-byte ASN>:<2-byte integer>"
+    """
+    branch = randgen.randint(0, 2)
+    if branch == 0:
+        # 16-bit ASNs 64496–64511 are reserved for documentation and sample code
+        return f"{randgen.randint(64496, 64511)}:{randgen.randint(0, 2**32 - 1)}"
+    if branch == 1:
+        return f"{faker.Faker().ipv4_private()}:{randgen.randint(0, 2**16 - 1)}"
```

Comment:
Any reason to use `randgen.randint` over `Faker("pyint")`?

Reply:
It's a bit shorter, compared to `faker.Faker().pyint(0, 2**16 - 1)`? :)

Not really - this was the best approach I came up with before you pointed out that I can use faker.Faker() directly as needed. I'll revisit this.